### PR TITLE
fix(PE-7163): update validate utils

### PR DESCRIPTION
**Pull request by atticusofsparta**

### CODE REVIEW

File: .luacheckrc [hunk 1]
```diff
@@ -1,6 +1,7 @@
 allow_defined = true
 exclude_files = {
 	"dist/",
+	"src/common/crypto"
```
Comment (fedellen) [inline .luacheckrc:4]: Are all these crypto files duplicated from IO contract? Or do they have any differences we should review

File: src/common/balances.lua [hunk 2]
```diff
@@ -7,9 +7,10 @@ local balances = {}
 
 --- Transfers the ANT to a specified wallet.
 ---@param to string - The wallet address to transfer the balance to.
+---@param allowUnsafeAddresses boolean Whether to allow unsafe addresses
```
Comment (fedellen) [inline src/common/balances.lua:10]: nit: can share type declarations and desriptions e.g:
 ```lua
 @alias AllowUnsafeAddresses boolean Whether to allow unsafe addresses
 
 
 @param allowUnsafeAddresses AllowUnsafeAddresses
 ```

File: src/common/controllers.lua [hunk 3]
```diff
@@ -18,9 +19,10 @@ end
 
 --- Remove a controller.
 ---@param controller string The controller to remove.
+---@param allowUnsafeAddresses boolean Whether to allow unsafe addresses
 ---@return string[]
-function controllers.removeController(controller)
-	utils.validateArweaveId(controller)
+function controllers.removeController(controller, allowUnsafeAddresses)
+	assert(utils.isValidAOAddress(controller, allowUnsafeAddresses), "Invalid AO Address")
 	local controllerExists = false
```
Comment (fedellen) [inline src/common/controllers.lua:27]: I'd imagine that we don't need to assert addresses on remove controller. If they're on the ANT as a controller with any address, the owner should be able to remove them

Comment (atticusofsparta) [reply src/common/controllers.lua:27]: agreed

Comment (atticusofsparta) [reply src/common/controllers.lua]: [a44924e](https://github.com/ar-io/ar-io-ant-process/pull/38/commits/a44924e9d1952b43bdc8ed53222edcb3d3403087)

I did assert at a minimum it must be a string

File: src/common/balances.lua (hunk 2 above)
Comment (atticusofsparta) [reply src/common/balances.lua]: [2242e41](https://github.com/ar-io/ar-io-ant-process/pull/38/commits/2242e41d9b0e15fb533569c5c493dc2a4042a6b1)

File: .luacheckrc (hunk 1 above)
Comment (atticusofsparta) [reply .luacheckrc:4]: I think they are duplicated, will replace just in case

Comment (atticusofsparta) [reply .luacheckrc:4]: Looks like they were indeed already duplicated.

File: src/common/utils.lua [hunk 4]
```diff
@@ -208,7 +268,20 @@ function utils.createHandler(tagName, tagValue, handler, position)
...
+			-- handling for eth EIP-55 format, returns address if is not eth address
+			msg.From = utils.formatAddress(msg.From)
+			local knownAddressTags = {
+				"Recipient",
+				"Controller",
+			}
+			for _, tName in ipairs(knownAddressTags) do
+				-- Format all incoming addresses
+				msg.Tags[tName] = msg.Tags[tName] and utils.formatAddress(msg.Tags[tName]) or nil
+				-- aos assigns tag values to the base message level as well
```
Comment (fedellen) [inline src/common/utils.lua:280]: interesting. so theres not really any need to explicitly reach into `Tags` object?

**Review by fedellen — APPROVED** [review]

**Pull request merged**